lib/rules/no-duplicate-selectors/index.cjs

```diff
@@ -53,122 +55,217 @@ const rule = (primary, secondaryOptions) => {
 		// from other rules that share the same parent and the same source.
 		const selectorContextLookup = nodeContextLookup();
 
-		root.walkRules((ruleNode) => {
-			if (isKeyframeRule(ruleNode)) {
-				return;
-			}
-
-			const contextSelectorSet = selectorContextLookup.getContext(
-				ruleNode,
-				findAtRuleContext(ruleNode),
-			);
-			const resolvedSelectorList = [
-				...new Set(
-					ruleNode.selectors.flatMap((selector) => resolvedNestedSelector(selector, ruleNode)),
-				),
-			];
-
-			const normalizedSelectorList = resolvedSelectorList.map(normalize);
+		/**
+		 * @param {import('postcss').Rule} node
```

Comment:
suggestion
refactor

You have got 4 `import('postcss').Rule`, maybe add a `/** @typedef {import('postcss').Rule} Foo */`?

Reply:
I've added this instead:

```js
/** @import { Rule as PostcssRule } from 'postcss' */
/** @import { Node as SelectorNode, Root as SelectorRoot, Selector } from 'postcss-selector-parser' */
```